tests/json_front_tests/test_json_class.py

```diff
@@ -0,0 +1,170 @@
+        self.assertEqual(ctf_meta[0].datatype, self.json_class.metas[0].datatype)
+
+    def test_argparse(self):
+        args = self.json_class.get_args(["--data_path=data.csv", "--dag_path=dag.dot", "--json_path=tests.json"])
+        self.assertTrue(args.data_path)
+        self.assertTrue(args.dag_path)
+        self.assertTrue(args.json_path)
+
+    def test_setup_modelling_scenario(self):
+        self.json_class.setup()
+        print(type(self.json_class.modelling_scenario))
+        print(self.json_class.modelling_scenario)
+        self.assertIsInstance(self.json_class.modelling_scenario, Scenario)
+
+    def test_setup_causal_specification(self):
+        self.json_class.setup()
+        self.assertIsInstance(self.json_class.causal_specification, CausalSpecification)
+
+    def test_abstract_test_case_generation(self):
+        self.json_class.setup()
+        effects = {"NoEffect": NoEffect()}
+        mutates = None
+        expected_effect = dict({"test_output": "NoEffect"})
+        example_test = {
+            "name": "test1",
+            "mutations": {},
+            "estimator": None,
+            "estimate_type": None,
+            "effect_modifiers": [],
+            "expectedEffect": expected_effect,
+            "skip": False,
+        }
+        abstract_test_case = self.json_class._create_abstract_test_case(example_test, mutates, effects)
+        self.assertIsInstance(abstract_test_case, AbstractCausalTestCase)
+
+    def test_execute_concrete_test(self):
+        self.json_class.setup()
+        dag_string = "digraph G { test_input -> B; B -> C; test_output -> test_input; test_output -> C}"
+        setup_dag_file(self.dag_path, dag_string)
+        effects = {"NoEffect": NoEffect()}
+        mutates = {
+            "Increase": lambda x: self.json_class.modelling_scenario.treatment_variables[x].z3 >
+                                  self.json_class.modelling_scenario.variables[x].z3
+        }
+        expected_effect = dict({"test_output": "NoEffect"})
+        example_test = {
+            "name": "test1",
+            "mutations": {"test_input": "Increase"},
+            "estimator": "LinearRegressionEstimator",
+            "estimate_type": "ate",
+            "effect_modifiers": [],
+            "expectedEffect": expected_effect,
+            "skip": False,
+        }
+        estimators = {"LinearRegressionEstimator": LinearRegressionEstimator}
+        abstract_test_case = self.json_class._create_abstract_test_case(example_test, mutates, effects)
+        concrete_tests, dummy = abstract_test_case.generate_concrete_tests(5, 0.5)
+
+        # This scenario should produce 5 failures
+        failures = self.json_class._execute_tests(concrete_tests, estimators, example_test, False)
+        self.assertEqual(failures, 5)
+
+    def tearDown(self) -> None:
+        remove_temp_dir_if_existent()
+
+
+def populate_example(*args, **kwargs):
+    pass
+
+
+def setup_json_file(json_path):
```

Comment:
Ah okay, good to know! Personally I prefer having it in the code as it is a relatively simple setup and it was useful for me to reference to when looking at the tests. 
Is this more of a preference rather than a 'right/wrong' trade-off then?

Reply:
I think file size / computational cost would heavily influence decision if they were issues.

Reply:
Makes sense thanks :). For now it is a very small json file with only a few lines in it.

The current setup creates and deletes this file for each test though. So I'll follow your suggestion as the more scalable and cleaner format! 

Cheers bob